misc/vtk/src/processors/vtktovolume.cpp

```diff
@@ -0,0 +1,406 @@
+    addProperty(boolProp);
+    numArrayComponents.push_back(numChannels);
+    ptrArrayData.push_back(reinterpret_cast<unsigned char *>(dataPtr));
+}
+
+void VTKtoVolume::VTKArrayList::clear() {
+    std::vector<std::string> identifiers{};
+    for (const auto &property : getProperties()) {
+        identifiers.emplace_back(property->getIdentifier());
+    }
+    for (const auto &identifier : identifiers) {
+        Property *prop = removeProperty(identifier);
+        delete prop;
+    }
+    numArrayComponents.clear();
+    ptrArrayData.clear();
+}
+
+void VTKtoVolume::updateFormats() {
+    NetworkLock lock;
+    reactToChange_ = false;
+
+    DataFormatId formerFormat = DataFormatId::NotSpecialized;
+    if (dataFormats_.size()) formerFormat = dataFormats_.getSelectedValue();
+
+    dataFormats_.clearOptions();
+    if (!inport_.hasData()) {
+        dataArrays_.clear();
+        return;
+    }
+
+    // Get dimensions. If none are given, it's not a rectilinear grid.
+    const VTKDataSet &vtkData = *inport_.getData();
+    std::optional<size3_t> dimensions = inport_.getData()->getDimensions();
+    if (!dimensions) {
+        LogWarn("Input vtk data set is not rectilinear.");
+        return;
+    }
+
+    // Get data attributes.
+    auto cellData = vtkData->GetCellData();
+    auto pointData = vtkData->GetPointData();
+    if ((useCellData_.get() && !cellData) || (!useCellData_.get() && !pointData)) return;
+    int numArrays =
+        useCellData_.get() ? cellData->GetNumberOfArrays() : pointData->GetNumberOfArrays();
+
+    // Find first non-internal data array and take that value type.
+    std::vector<int> insertedValues;
+    for (int i = 0; i < numArrays; ++i) {
+        auto array = useCellData_.get() ? cellData->GetArray(i) : pointData->GetArray(i);
+        int vtkTypeId = array->GetDataType();
+
+        if (std::find(insertedValues.begin(), insertedValues.end(), vtkTypeId) !=
+            insertedValues.end())
+            continue;
+
+        const DataFormatBase *format =
+            getFormatFromVtkId(vtkTypeId, array->GetElementComponentSize() * 8);
+        if (!format) continue;
+
+        insertedValues.push_back(vtkTypeId);
+        dataFormats_.addOption(format->getString(), format->getString(), format->getId());
+    }
+
+    // Try to keep arrays if the new input has arrays of the same type.
+    bool keep = dataFormats_.setSelectedValue(formerFormat);
+
+    if (dataFormats_.size())
+        updateArrays(keep);
+    else
+        dataArrays_.clear();
+
+    reactToChange_ = true;
+}
+
+void VTKtoVolume::updateArrays(bool keepSettings) {
+    NetworkLock lock;
+    reactToChange_ = false;
+
+    // Remember selected channels
+    std::unordered_set<std::string> formerChannels;
+    if (keepSettings) {
+        if (!formerArraySelection_.empty()) {
+            for (auto &name : formerArraySelection_) formerChannels.insert(name);
```

Comment:
std::copy

Reply:
Changed to std::vector and simply assigned. Less overhead form data structure for <= 4 elements anyway.